### PR TITLE
Nico - Updated Backend to Spring Boot 3

### LINKED ISSUE #8: Update Backend to Spring 3
In this issue, we'll update the backend code to Spring 3

# Why we need to do this.

* Security; some of the dependencies we are using have security issues that are only patched in Spring 3.  (This is the main reason)
* Features; Spring 3 has some capabilities that are not present in Spring 2. (At moment, none of them are critical, but it would be better to be on the version of Spring that is getting more updates.)

# How to do it

This PR shows how we updated the code base for STARTER-team03 to Spring 3.  It is likely that if you do the exact same things, it will work for this code base:

* <https://github.com/ucsb-cs156-f24/STARTER-team03/pull/13>

# Notes

* Q: Do I need the aop class that is in the STARTER-team03 code?
  * A: No, it's not necessary. However, that's also a nice thing to add, and we'll make a separate issue for that.
* Q: The STARTER-team03 has code for integration and end2end tests.  Do we need to add those in this issue?
  * A: No.  However, adding the lines in the pom.xml that exclude those from pitest is still a good idea, since those might get added into the code base in the future.

### PR DESCRIPTION
**Pull request by nicoshinozaki**
In this PR, we upgrade the backend to Spring Boot 3.

Deployed at: http://organic-nicoshinozaki-dev.dokku-10.cs.ucsb.edu

Closes #8

### CODE REVIEW

File: src/main/java/edu/ucsb/cs156/organic/config/SecurityConfig.java [hunk 1]
```diff
@@ -55,25 +78,25 @@ public class SecurityConfig extends WebSecurityConfigurerAdapter {
   // @Autowired
   // private LoginSuccessHandler loginSuccessHandler;
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/organic/config/SecurityConfig.java]: Please remove the commented out code (yes, I know it was in the original, but the rule is, if you touch the file, you have to clean it up.)

**Review by pconrad — CHANGES_REQUESTED** [review]
LGTM except for a couple of things:
* Please update the description
* Please remove some commented out code.
* Please deploy to a dokku dev instance and link to the deployed instance in the PR description.  This is extra important because of the fact that we don't yet have full integration and end2end tests in this code base yet, and so the only way to know that we didn't break anything is to test it manually.
* Please consider making the change that Daniel Jensen suggested on the slack <https://ucsb-cs156-s24.slack.com/archives/C074MLEP1QD/p1716436359312639>

**Review by abhishekambastha04 — APPROVED** [review]
LGTM

**Review by pconrad — APPROVED** [review]
LGTM!

**Pull request merged**